proposals/0000-severe-warnings.rst

```diff
@@ -0,0 +1,214 @@
+
+  It behaves like the ``compat`` group in that ``-Wno-severe``, ``-Wwarn=severe`` and ``-Werror=severe`` behave as if the flag
+  had been specified for each warning in the groups.
+
+  Warning categories in this group are on and erroring by default (as if ``-Werror=severe`` was the first argument).
+  
+* The warning group is initialized with the following warning categories:
+
+  * ``missing-methods``
+  * ``missing-fields``
+
+  They are no longer part of ``-Wdefault``.
+
+*  The set of allowed names of custom warning categories (`#541 <https://github.com/ghc-proposals/ghc-proposals/pull/541>`_) is
+   extended by those prefixed with ``xs-*`` (for “custom extended – severe”). The flags
+   ``-Wno-severe``, ``-Wwarn=severe`` and ``-Werror=severe``
+   also affect all custom warnings with a name starting in ``xs-*``.
+
+* For uniformity, the warning groups ``default`, ``extra``, ``all`` and ``everything`` can also be used in ``-Wno-<group>``,
+  ``-Wwarn=<group>``  and ``-Werror=<group>``. (This is already implemented in `MR9679 <https://gitlab.haskell.org/ghc/ghc/-/merge_requests/9679>`_).
+
+Examples
+--------
+
+Consider file ``Test.hs``::
+
+  module Test where
+  data Foo = Foo { a :: Int, b :: Bool }
+  foo = Foo { a = 1 }
+
+We would get::
+
+  $ ghc Test.hs
+  [1 of 1] Compiling Test             ( Test.hs, Test.o )
+  
+  Test.hs:3:9: error: [-Wmissing-fields, -Werror=missing-fields]
+      • Fields of ‘Foo’ not initialised:
+          b :: Bool
+      • In the expression: Foo {a = 1}
+        In an equation for ‘foo’: foo = Foo {a = 1}
+    |
+  3 |   foo = Foo { a = 1 }
+    |         ^^^^^^^^^^^^^
+  $ echo $?
+  1
+  $ ghc -Wwarn=missing-fields Test.hs
+  [1 of 1] Compiling Test             ( Test.hs, Test.o )
+  
+  Test.hs:3:9: warning: [-Wmissing-fields]
+      • Fields of ‘Foo’ not initialised:
+          b :: Bool
+      • In the expression: Foo {a = 1}
+        In an equation for ‘foo’: foo = Foo {a = 1}
+    |
+  3 |   foo = Foo { a = 1 }
+    |         ^^^^^^^^^^^^^
+  $ echo $?
+  0
+  $ ghc -Wno-missing-fields Test.hs
+  $ echo $?
+  0
+
+Disabling the ``default`` warnings, but not the ``severe`` warnings, will leave the ``severe`` warnings on::
+
+  $ ghc Test.hs -Wno-default
+  [1 of 1] Compiling Test             ( Test.hs, Test.o )
+  
+  Test.hs:3:9: error: [-Wmissing-fields, -Werror=missing-fields]
+      • Fields of ‘Foo’ not initialised:
+          b :: Bool
+      • In the expression: Foo {a = 1}
+        In an equation for ‘foo’: foo = Foo {a = 1}
+    |
+  3 |   foo = Foo { a = 1 }
+    |         ^^^^^^^^^^^^^
+  
+
+Effect and Interactions
+-----------------------
+None yet.
+
+
+Costs and Drawbacks
+-------------------
+In terms of GHC development, this is a modest extension of the existing warning category and group infrastructure.
+
+In terms of breakage, @phadej has made an impact analysis at <https://github.com/ghc-proposals/ghc-proposals/issues/544#issue-1410125536>.
```

Comment:
This comment talks about `missing-methods`. It wasn't immediately obvious to me that he also kindly provided an analysis for `missing-fields` (https://github.com/ghc-proposals/ghc-proposals/issues/544#issuecomment-1279948737), so perhaps link to that here?

Given the likely impact I think the proposal text could do with backporting the "Backwards Compatibility" section from the latest version of the template, explaining that (a) there will be breakage and (b) the breakage is justified.

Reply:
I added the section with, very little text. If you want, feel free to elaborate (just commit directly, or use the suggested edit feature :-))

Reply:
Looks good. :-) Ready to submit?